package/MDAnalysis/transformations/boxdimensions.py

```diff
@@ -25,44 +25,63 @@
 Set box dimensions --- :mod:`MDAnalysis.transformations.boxdimensions`
 =======================================================================
 
-Set dimensions of the simulation box to a constant vector across all timesteps.
-
+Set dimensions of the simulation box, either to a constant vector across
+all timesteps or to a specified vector at each frame.
 
 .. autoclass:: set_dimensions
 """
 import numpy as np
 
 from .base import TransformationBase
 
-
 class set_dimensions(TransformationBase):
     """
     Set simulation box dimensions.
 
     Timestep dimensions are modified in place.
 
-    Example
-    -------
+    Examples
+    --------
 
     e.g. set simulation box dimensions to a vector containing unit cell
     dimensions [*a*, *b*, *c*, *alpha*, *beta*, *gamma*], lengths *a*,
     *b*, *c* are in the MDAnalysis length unit (Å), and angles are in degrees.
+    The same dimensions will be used for every frame in the trajectory.
 
     .. code-block:: python
 
-        dim = [2, 2, 2, 90, 90, 90]
+        dim = np.array([2, 2, 2, 90, 90, 90])
+        transform = mda.transformations.boxdimensions.set_dimensions(dim)
+        u.trajectory.add_transformations(transform)
+
+    Or e.g. set simulation box dimensions to a vector containing unit cell
+    dimensions [*a*, *b*, *c*, *alpha*, *beta*, *gamma*] at the first frame,
+    and [*2a*, *2b*, *2c*, *alpha*, *beta*, *gamma*] at the second frame.
+
+    .. code-block:: python
+
+        dim = np.array([
+            [2, 2, 2, 90, 90, 90],
+            [4, 4, 4, 90, 90, 90],
+        ])
         transform = mda.transformations.boxdimensions.set_dimensions(dim)
         u.trajectory.add_transformations(transform)
 
     Parameters
     ----------
-    dimensions: iterable of floats
+    dimensions: iterable of floats or two-dimensional np.typing.NDArrayLike
         vector that contains unit cell lengths and angles.
-        Expected shapes are (6, 0) or (1, 6)
+        Expected shapes are (6, 0) or (1, 6) or (N, 6), where
+        N is the number of frames in the trajectory. If shape is (6, 0) or
+        (1, 6), the same dimensions will be used at every frame in the
+        trajectory.
 
     Returns
     -------
     :class:`~MDAnalysis.coordinates.timestep.Timestep` object
+
+    .. versionchanged:: 2.7.0
```

Comment:
```suggestion

    .. versionchanged:: 2.7.0
```
This is one of those annoying ones where docs don't render well if you don't have enough empty lines, see: https://mdanalysis--4258.org.readthedocs.build/en/4258/documentation_pages/transformations/boxdimensions.html

Reply:
This still needs addressing from the looks of things.

Reply:
Woops, sorry I missed that